### PR TITLE
Port from_timestamp_millis() (#818) to 0.4.x

### PR DESCRIPTION
**Pull request by djc**
Added some changes to deduplicate logic in the code.

### CODE REVIEW

File: src/naive/datetime/mod.rs [hunk 1]
```diff
@@ -131,6 +135,39 @@ impl NaiveDateTime {
...
+    /// let timestamp_millis: i64 = -2208936075; //Mon Jan 01 1900 14:38:45 GMT+0000
+    /// let naive_datetime = NaiveDateTime::from_timestamp_millis(timestamp_millis);
+    /// assert!(naive_datetime.is_some());
+    /// assert_eq!(timestamp_millis, naive_datetime.unwrap().timestamp_millis());
+    /// ```
+    #[inline]
+    pub fn from_timestamp_millis(millis: i64) -> Option<NaiveDateTime> {
+        let mut secs = millis / 1000;
+        if millis < 0 {
+            secs = secs.checked_sub(1)?;
```
Comment (esheppa) [inline src/naive/datetime/mod.rs:163]: I think we still need to include this part as well: `NANOS_IN_SECOND.checked_sub(nsecs)?,` (in the `millis < 0` case only)

**Review by esheppa — CHANGES_REQUESTED** [review]

File: src/naive/datetime/mod.rs (hunk 1 above)
Comment (djc) [reply src/naive/datetime/mod.rs:165]: Oh yeah, good catch. Added a commit that enables CI to run on the 0.4.x branch, too...

**Review by esheppa — APPROVED** [review]
Looks good! Maybe backporting to `0.4.x` is cleaner - it keeps the history of the `main` branch nicer as well

**Comment by djc** [comment]
Hmm, let's try a bit more -- we should be a little more forceful about getting authors to target 0.4.x, I think.

**Comment by Pscheidl** [comment]
This version is cleaner, but should be slightly less efficient - two ifs instead of one and the 'abs' operation is always performed. Not sure if you care about the difference, just explaining the original implementation.

**Comment by djc** [comment]
Yeah -- unless someone wants to build benchmarks that show a substantial improvement, I'd rather opt for code that's easier to follow.

As for the `abs()` call -- since `abs()` will internally call `is_negative()` anyway, branching in order to prevent an `abs()` call doesn't actually help -- it will have the same number of branches in the case it's not needed, and an extra branch in the case it is needed.

**Pull request merged**